sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/logical/statsEstimation/UnionEstimation.scala

```diff
@@ -111,6 +111,44 @@ object UnionEstimation {
       AttributeMap.empty[ColumnStat]
     }
 
+    val attrToComputeNullCount = union.children.map(_.output).transpose.zipWithIndex.filter {
+      case (attrs, _) => attrs.zipWithIndex.forall {
+        case (attr, childIndex) =>
+          val attrStats = union.children(childIndex).stats.attributeStats
+          attrStats.get(attr).isDefined && attrStats(attr).nullCount.isDefined
+      }
+    }
+
+    val newAttrStats = if (attrToComputeNullCount.nonEmpty) {
+      val outputAttrStats = new ArrayBuffer[(Attribute, ColumnStat)]()
+      attrToComputeNullCount.foreach {
+        case (attrs, outputIndex) =>
+          val colWithNullStatValues = attrs.zipWithIndex.foldLeft[Option[BigInt]](None) {
+            case (totalNullCount, (attr, childIndex)) =>
+              val colStat = union.children(childIndex).stats.attributeStats(attr)
+              if (totalNullCount.isDefined) {
```

Comment:
`totalNullCount.isDefined` can be false?

Reply:
Yes, initial value of `totalNullCount` is None.

Reply:
If the first element can be null only, could we remove this if?
```
          val firstStat = union.children.head.stats.attributeStats(attrs.head)
          val firstNullCount = firstStat.nullCount.get
          attrs.zipWithIndex.tail.foldLeft(firstNullCount) {...}
```

Reply:
Done.